src/presentation/http/router/note.test.ts

```diff
@@ -159,6 +159,53 @@ describe('Note API', () => {
       }
     });
 
+    test('Returns 200 if user is team member of parent note with Write role', async () => {
+      const creator = await global.db.insertUser();
+
+      const randomGuy = await global.db.insertUser();
+
+      const note = await global.db.insertNote({
+        creatorId: creator.id,
+      });
+
+      await global.db.insertNoteSetting({
+        noteId: note.id,
+        isPublic: false,
+      });
+
+      const parentNote = await global.db.insertNote({
+        creatorId: creator.id,
+      });
+
+      await global.db.insertNoteRelation({
+        noteId: note.id,
+        parentId: parentNote.id,
+      });
+
+      await global.db.insertNoteTeam({
+        noteId: parentNote.id,
+        userId: randomGuy.id,
+        role: MemberRole.Write,
```

Comment:
what about read role? Maybe we need to use `test.each` way to check all roles?

Reply:
done